### PR TITLE
CAS-420  -  Update strategy selector to use react-hook-form

### PR DESCRIPTION
**Pull request by germanurrus**
This PR will be merged to the feature branch https://github.com/DapperCollectives/CAST/pull/349
- Adds react hook form to strategy selector
To test create a community and use the strategy with the information required

### CODE REVIEW

File: frontend/packages/client/src/components/Community/CommunityEditorDetails/FormConfig.js [hunk 1]
```diff
@@ -99,13 +103,13 @@ const AddressSchema = ({ isValidFlowAddress, isEditMode = false } = {}) => {
...
-        }),
-  });
+  return yup
+    .object()
+    .shape(
+      Object.assign(
+        {},
+        ...fieldNames.map((fieldName) => ({ [fieldName]: elementValidation }))
+      )
+    );
```
Comment (germanurrus) [inline frontend/packages/client/src/components/Community/CommunityEditorDetails/FormConfig.js:113]: using `fieldNames` as param enables having in the same form fields that are arrays like admin addresses and authors then useArrayField will be used for each one of them and `elementValidation` will be applied

File: frontend/packages/client/src/components/Community/CommunityEditorDetails/MembersEditor.js [hunk 2]
```diff
@@ -38,7 +38,11 @@ export default function MembersEditor({
   const { register, control, handleSubmit, reset, formState } = useForm({
     mode: 'all',
     resolver: yupResolver(
-      AddressSchema({ isValidFlowAddress, isEditMode: true })
+      AddressSchema({
+        fieldNames: ['addrList'],
```
Comment (germanurrus) [inline frontend/packages/client/src/components/Community/CommunityEditorDetails/MembersEditor.js:42]: only one field array with name `addrList`

File: frontend/packages/client/src/components/Community/StrategyEditorModal/index.js [hunk 3]
```diff
@@ -26,25 +24,6 @@ const getFormFields = (strategy) => {
...
-  };
-
-  if (strategy === 'float-nfts') {
-    requiredFields.floatEventId = (floatEventId) =>
-      floatEventId?.trim().length > 0 && /^[0-9]+$/.test(floatEventId);
-  }
-
-  return requiredFields;
-};
-
```
Comment (germanurrus) [inline frontend/packages/client/src/components/Community/StrategyEditorModal/index.js:47]: validation not required

**Review by jbluks — APPROVED** [review]

**Comment by jbluks** [comment]
LGTM

**Pull request merged**